lib/pbench/test/unit/server/test_datasets_list.py

```diff
@@ -496,3 +496,48 @@ def do_error(
             if key in m:
                 assert error in m
                 break
+
+    def test_key_summary(self, query_as):
+        """Test keyspace summary.
+
+        With the `keysummary` query parameter, /datasets returns an aggregation
+        of defined metadata key namespaces for the selected datasets.
+
+        We add a few metadata kays to the ones provided by the fixture to show
+        aggregation across multiple selected datasets. Note that without filter
+        criteria, the query here should return drb's "drb" and "fio_1" datasets
+        and test's public "fio_2" dataset.
+        """
+        fio_1 = Dataset.query(name="fio_1")
+        Metadata.setvalue(dataset=fio_1, key="server.origin", value="SAT")
+        Metadata.setvalue(dataset=fio_1, key="global.legacy.server", value="ABC")
+        response = query_as({"keysummary": "true"}, "drb", HTTPStatus.OK)
```

Comment:
You don't want to see (disparate) contributions from multiple datasets?  (I.e., this use to have different keys from `fio_1` and `fio_2`, and now it's all coming from `fio_1`, right?)